src/opcua-mock-plc/simulation.ts

```diff
@@ -6,53 +6,72 @@ enum State {
   ERROR,
 }
 
-export const simulateMachine = (device: UAObject, namespace: Namespace): void => {
+export const simulateMachine = (device: UAObject, namespace: Namespace): (() => void) => {
```

Comment:
Why not reuse the existing simulation?

Reply:
this is a bigger rework in the matter, that error and errorDescription is added. The mock-machine-connector is creating events. To make it work, we have to add an interface there as well to keep them separated. 

Nice thing but requires some extra effort.

Reply:
I opened an Issue #3